web/src/main/java/org/mskcc/smile/web/SampleController.java

```diff
@@ -64,6 +71,37 @@ public ResponseEntity<List<PublishedSmileSample>> fetchSampleMetadataListByCmoPa
                 .body(samples);
     }
 
+    /**
+     * Given an input date as yyyy-MM-dd, returns a list of sample id mappings.
+     * @param importDate
+     * @return ResponseEntity
+     * @throws Exception
+     */
+    @ApiOperation(value = "Fetch SmileSampleIdMapping list by inputDate",
+        nickname = "fetchSmileSampleIdMappingListByInputDateGET")
+    @RequestMapping(value = "/samplesByDate/{importDate}",
+        method = RequestMethod.GET,
+        produces = "application/json")
+    public ResponseEntity<List<SmileSampleIdMapping>> fetchSampleIdMappingsByInputDateGET(
+            @ApiParam(value = "Import date to search from", required = true)
+            @PathVariable String importDate) throws Exception {
+        // validate input date string before submitting db query
+        try {
+            IMPORT_DATE_FORMATTER.parse(importDate);
```

Comment:
Do you have to capture the return of this call to parse?

importDate = IMPORT_DATA_FORMATTER.parse(importDate)